core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java

```diff
@@ -351,10 +352,22 @@ protected void runReplicateTest(IndexShardRoutingTable shardRoutingTable, int as
         internalRequest.concreteIndex(shardId.index().name());
         Releasable reference = getOrCreateIndexShardOperationsCounter();
         assertIndexShardCounter(2);
+        AtomicBoolean noMasterOccurred = new AtomicBoolean();
+        AtomicBoolean criticalFailureOccurred = new AtomicBoolean();
         TransportReplicationAction<Request, Request, Response>.ReplicationPhase replicationPhase =
                 action.new ReplicationPhase(shardIt, request,
                         new Response(), new ClusterStateObserver(clusterService, logger),
-                        primaryShard, internalRequest, listener, reference);
+                        primaryShard, internalRequest, listener, reference, new ShardStateActionListener() {
+                    @Override
```

Comment:
strictly speaking this testing belongs the tests for ShardStateAction , right? ReplicationAction doesn't do much with this now. Later one will be a different story, but we can the rely on the listener to work correctly and test how ReplicationAction responds. Or am I missing something?